### PR TITLE
add raw source support and resolve file path

### PR DESCRIPTION
**Pull request by dushaobindoudou**
add raw source support and resolve file path

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -268,6 +276,10 @@ HtmlWebpackIncludeAssetsPlugin.prototype.apply = function (compiler) {
         includeAssetCount = includeAssetPaths.length;
         for (var a = 0; a < includeAssetCount; a++) {
           includeAssetPath = includeAssetPaths[a];
+
+          // var fsContent = fs.readFileSync(includeAssetPath);
```
Comment (jharris4) [inline index.js]: Please remove commented out code :-)

This kind of code would be best to include in a new unit test...

**Comment by jharris4** [comment]
Hi! Thanks for the PR, this looks like a useful feature!

Do you think you could take a minute to add a unit test for this new feature though?

Adding unit tests helps both make sure this code works correctly, and also gives a kind of documentation for how this feature works.

**Comment by jharris4** [comment]
I removed the raw source feature since it didn't seem to be doing anything.

But I did add a new plugin option so you can do `{ resolvePaths: true }` if you want your assets paths to be resolved to absolute paths.

This was released in version `1.0.5`

I'd be happy to look into the `assets.rawData = ...` code you added again if you can explain why you added it.

**Comment by dushaobindoudou** [comment]
sometimes we need to add some inline code to the html so i add the rawData field

**Comment by jharris4** [comment]
Ok but how does it work? I couldn’t find rawData mentioned anywhere else...

**Comment by dushaobindoudou** [comment]
we use custom templates.  

`<% for (var css in htmlWebpackPlugin.files.css) { %>
            <% if(htmlWebpackPlugin.files.rawData[htmlWebpackPlugin.files.css[css].substr(htmlWebpackPlugin.files.publicPath.length)]) {%>
            <style>
                <%= htmlWebpackPlugin.files.rawData[htmlWebpackPlugin.files.css[css].substr(htmlWebpackPlugin.files.publicPath.length)].source %>
            </style>
            <% } %>
        <% } %>`

this can work

**Comment by jharris4** [comment]
Ok, I see what you're getting at. It made me wonder if `html-webpack-plugin` supports inline content directly and a quick search found this:

https://github.com/jantimon/html-webpack-plugin/issues/104

I'm curious if any of the solutions listed there would also work for you?

**Comment by dushaobindoudou** [comment]
We need to add additional asset resources, not all cases need inline, in most cases still use src / href way, so Improved your plugin。I think this plugin supports raw is not a bad thing, the plugin can support more scenarios

**Comment by jharris4** [comment]
Ok, but new features need documentation and tests! 😉

I’m trying to make sure I understand the feature you added enough to add those for it...

**Comment by dushaobindoudou** [comment]
I will resubmit a pull request, including tests and documentation

**Pull request merged**